BaseTools/Source/C/GenFw/GenFw.c

```diff
@@ -443,6 +443,64 @@ Routine Description:
   return STATUS_SUCCESS;
 }
 
+// MU_CHANGE begin
+BOOLEAN
+IsNxCompatCompliant (
+  EFI_IMAGE_OPTIONAL_HEADER_UNION  *PeHdr
+  )
+/*++
+
+Routine Description:
+
+  Checks if the Pe image is nxcompat. i.e. PE is 64bit, section alignment is
+  divisible by 4k, and all sections are not Write/Execute
+
+Arguments:
+
+  PeHdr      The Pe header
+
+Returns:
+
+  TRUE       The PE is nx compat compliant
+  FALSE      The PE is not nx compat compliant
+--*/
+{
+  EFI_IMAGE_SECTION_HEADER     *SectionHeader;
+  EFI_IMAGE_OPTIONAL_HEADER64  *Optional64;
+  UINT32                       Index;
+  UINT32                       Mask;
+
+  // Must have an optional header to perform verification
+  if (PeHdr->Pe32.FileHeader.SizeOfOptionalHeader == 0) {
+    return FALSE;
+  }
+
+  // Verify PE is 64 bit
+  if (!(PeHdr->Pe32.OptionalHeader.Magic == EFI_IMAGE_NT_OPTIONAL_HDR64_MAGIC)) {
+    return FALSE;
+  }
+
+  // Verify Section Alignment is divisible by 4K
+  Optional64 = (EFI_IMAGE_OPTIONAL_HEADER64 *)&PeHdr->Pe32.OptionalHeader;
+  if (!((Optional64->SectionAlignment % 0x1000) == 0)) {

```

Comment:
Given this is targeting edk2 build tools, `EFI_PAGE_SIZE` should be used. If someone substituted the page size value, this would be one less place they have to update to use it for things to work as expected.

```suggestion
  if (!((Optional64->SectionAlignment % EFI_PAGE_SIZE) == 0)) {

```

Reply:
Awesome, Thank you!